File: 2824-count-pairs-whose-sum-is-less-than-target/2824-count-pairs-whose-sum-is-less-than-target.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number}
 */
var countPairs = function(nums, target) {
    nums.sort((a, b) => a - b)
    let l = 0, r = nums.length-1, res = 0
    while (nums[l] + nums[r] > target) r--
    for (let i=l; i <= r; i++) {
        for (let j=i+1; j <= r; j++) {
            if (nums[i] + nums[j] < target) res++
        }
    }
    return res
};

File: 2824-count-pairs-whose-sum-is-less-than-target/2824-count-pairs-whose-sum-is-less-than-target.py
class Solution:
    def countPairs(self, nums: List[int], target: int) -> int:
        nums.sort()
        l, r, res = 0, len(nums)-1, 0
        while l < r:
            if nums[l] + nums[r] < target:
                res += r - l
                l += 1
            else:
                r -= 1
        return res